boat/memberlog.js

```diff
@@ -0,0 +1,155 @@
+    bot.on('guildMemberRemove', (guild, member) => this.memberRemove(bot, guild, member))
+    bot.on('guildMemberUpdate', (guild, member, oldMember) => this.memberUpdate(bot, guild, member, oldMember))
+  },
+
+  memberAdd (bot, guild, member) {
+    if (guild.id !== config.discord.ids.serverId) return
+    const createdAt = new Date(member.createdAt)
+    const now = new Date()
+
+    const embed = {
+      title: `${member.username}#${member.discriminator} just joined`,
+      description: `<@${member.id}> created their accout at ${createdAt.toUTCString()} (${humanTime(now - createdAt)})`,
+      timestamp: now.toISOString(),
+      color: 0x7289da,
+      thumbnail: { url: member.avatarURL },
+      footer: { text: `Discord ID: ${member.id}` }
+    }
+    bot.createMessage(memberLog, { embed })
+  },
+
+  memberRemove (bot, guild, member) {
+    if (guild.id !== config.discord.ids.serverId) return
+    const now = Date.now()
+    let description = `<@${member.id}> was not in the cache when they left`
+
+    if (member.joinedAt) {
+      const joinedAt = new Date(member.joinedAt)
+      description = `<@${member.id}> was here for ${humanTime(now - joinedAt)}`
+    }
+
+    const embed = {
+      title: `${member.user.username}#${member.user.discriminator} just left`,
+      description,
+      color: 0xdac372,
+      thumbnail: { url: member.user.avatarURL },
+      footer: { text: `Discord ID: ${member.user.id}` }
+    }
+    bot.createMessage(memberLog, { embed })
+  },
+
+  memberUpdate (bot, guild, newMember, oldMember) {
+    if (guild.id !== config.discord.ids.serverId) return
+
+    let embed = {
+      title: `An external update happened to ${newMember.username}#${newMember.discriminator}`,
+      color: 0xbdda72,
+      thumbnail: { url: newMember.avatarURL },
+      footer: { text: `Discord ID: ${newMember.id}` }
+    }
+
+    if (oldMember.nick !== newMember.nick) embed = this.nickChange(newMember, oldMember.nick)
+    if (newMember.roles.length !== oldMember.roles.length || !newMember.roles.every(role => oldMember.roles.includes(role))) embed = this.roleChange(newMember, oldMember)
+
+    bot.createMessage(memberLog, { embed })
+  },
+
+  nickChange (member, oldNick) {
+    return {
+      title: `${member.username}#${member.discriminator} changed their nickname`,
+      color: 0x8f72da,
+      fields: [
+        {
+          name: 'New nickname',
+          value: member.nick ? member.nick : member.username,
+          inline: true
+        }, {
+          name: 'Old nickname',
+          value: oldNick || member.username,
+          inline: true
+        }
+      ],
+      thumbnail: { url: member.avatarURL },
+      footer: { text: `Discord ID: ${member.id}` }
+    }
+  },
+
+  roleChange (newMember, oldMember) {
+    const newRoleIDs = newMember.roles.filter(this._comparer(oldMember.roles))
+    const oldRoleIDs = oldMember.roles.filter(this._comparer(newMember.roles))
```

Comment:
```suggestion
    const newRoleIDs = newMember.roles.filter(r => !oldMember.roles.includes(r))
    const oldRoleIDs = oldMember.roles.filter(r => !newMember.roles.includes(r))
```
Might also be worth renaming the variables to "addedRoleIds" and "removedRoleIds", since it's imo a better naming choice.

It's also not enforced by the linter but I prefer strictCamelCase, which means `somethingId` shouldn't be named `somethingID`. The camelCase rule goes over the rule of having abbreviations in uppercase.